### PR TITLE
test(core): refactor to simplify test configuration overrides

### PR DESCRIPTION
**Pull request by ideoma**
Motivation:
Do not write 1000 lines of code to test non-default property settings and remove 1000 lines of existing plumbing.

Test property override:
Any configuration property can be overridden in tests with 

```Java
// Static setup
setProperty(PropertyKey.CAIRO_SNAPSHOT_INSTANCE_ID, snapshotId);

// From the tests or non-static setup
node1.setProperty(PropertyKey.CAIRO_SNAPSHOT_INSTANCE_ID, snapshotId);
```

No plumbing code is needed to override a config value.

Also, this PR makes the tests run in random order by adding 

```Java
@OrderWith(RandomOrder.class)
public abstract class AbstractCairoTest extends AbstractTest {
```

It uncovered a few test bugs and a bug in `IntStack` impacting `FunctionParaser`. These are fixed in the PR.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 231 / 252 (91.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 234 / 255 (91.76%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 240 / 262 (91.60%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 240 / 262 (91.60%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 240 / 262 (91.60%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|

File: core/src/test/java/io/questdb/test/griffin/engine/functions/catalogue/ShowTablesFunctionFactoryTest.java [hunk 1]
```diff
@@ -57,8 +58,8 @@ public void testMetadataQuery() throws Exception {
 
     @Test
     public void testMetadataQueryDefaultHysteresisParams() throws Exception {
-        configOverrideMaxUncommittedRows(83737);
-        configOverrideO3MaxLag();
+        node1.setProperty(PropertyKey.CAIRO_MAX_UNCOMMITTED_ROWS, 83737);
+        node1.setProperty(PropertyKey.CAIRO_O3_MAX_LAG, 28);
```
Comment (jerrinot) [inline core/src/test/java/io/questdb/test/griffin/engine/functions/catalogue/ShowTablesFunctionFactoryTest.java:62]: is `28` a typo? the old method uses the value `28291`.
```
    protected static void configOverrideO3MaxLag() {
        node1.getConfigurationOverrides().setO3MaxLag(28291);
    }
```

File: core/src/test/java/io/questdb/test/griffin/O3SquashPartitionTest.java [hunk 2]
```diff
@@ -786,8 +803,10 @@ private void testSquashPartitionsOnEmptyTable(String wal) throws Exception {
     private void testSquashPartitionsOnNonEmptyTable(String wal) throws Exception {
         assertMemoryLeak(() -> {
             // 4kb prefix split threshold
-            node1.getConfigurationOverrides().setPartitionO3SplitThreshold(4 * (1 << 10));
-            node1.getConfigurationOverrides().setO3PartitionSplitMaxCount(2);
+            Overrides overrides1 = node1.getConfigurationOverrides();
+            overrides1.setProperty(PropertyKey.CAIRO_O3_PARTITION_SPLIT_MIN_SIZE, 4 * (1 << 10));
+            Overrides overrides = node1.getConfigurationOverrides();
```
Comment (jerrinot) [inline core/src/test/java/io/questdb/test/griffin/O3SquashPartitionTest.java:808]: nit - why are we getting `overrides` twice?

File: core/src/test/java/io/questdb/test/cutlass/line/tcp/LineTcpCommitFuzzTest.java [hunk 3]
```diff
@@ -101,14 +102,14 @@ public void testCommitIntervalBasedFraction() throws Exception {
...
-        configOverrideMaxUncommittedRows(500_000);
+        node1.setProperty(PropertyKey.CAIRO_MAX_UNCOMMITTED_ROWS, 500_000);
 
         // idle table commit after 5 mins inactivity -> test would timeout
         maintenanceInterval = 300_000_000;
         minIdleMsBeforeWriterRelease = 300_000_000;
 
         // time based commit every 0.2 seconds (20% of 1 sec commit lag) -> should commit rows -> make test pass
-        configOverrideO3MinLag(1_000_000);
+        node1.setProperty(PropertyKey.CAIRO_O3_MIN_LAG, 1_000);
```
Comment (jerrinot) [inline core/src/test/java/io/questdb/test/cutlass/line/tcp/LineTcpCommitFuzzTest.java:112]: is the value intentionally different?

File: core/src/test/java/io/questdb/test/griffin/engine/functions/catalogue/ShowTablesFunctionFactoryTest.java (hunk 1 above)
Comment (jerrinot) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/catalogue/ShowTablesFunctionFactoryTest.java:62]: ok, different units.

**Comment by jerrinot** [comment]
I reviewed a big chunk of this PR, but there are still parts to go through—apologies for this taking too long.

File: core/src/test/java/io/questdb/test/griffin/O3SquashPartitionTest.java (hunk 2 above)
Comment (ideoma) [reply core/src/test/java/io/questdb/test/griffin/O3SquashPartitionTest.java:808]: That's just debris of auto inlining, I can simplify it

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 240 / 262 (91.60%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 242 / 264 (91.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/RebuildColumnBase.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 242 / 264 (91.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SqlJitMode.java|3|5|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|189|208|90.87%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/IntStack.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/RebuildColumnBase.java|2|2|100.00%|

**Pull request merged**